contracts/LONStaking.sol

```diff
@@ -294,12 +315,24 @@ contract LONStaking is ERC20ForUpgradeable, OwnableForUpgradeable, ReentrancyGua
         uint256 share = balanceOf(msg.sender);
         if (block.timestamp > cooldownEndTimestamp) {
             // Normal redeem if cooldown period already passed
-            _redeem(share, 0);
+            _redeemAndSendTo(share, 0, msg.sender);
         } else {
             uint256 timeDiffInDays = Math.min(COOLDOWN_IN_DAYS, (cooldownEndTimestamp.sub(block.timestamp)).div(86400).add(1));
             // Penalty = share * (number_of_days_to_cooldown_end / number_of_days_in_cooldown) * (BPS_RAGE_EXIT_PENALTY / BPS_MAX)
             uint256 penalty = share.mul(timeDiffInDays).mul(BPS_RAGE_EXIT_PENALTY).div(BPS_MAX).div(COOLDOWN_IN_DAYS);
-            _redeem(share.sub(penalty), penalty);
+            _redeemAndSendTo(share.sub(penalty), penalty, msg.sender);
         }
     }
+
+    function convertXLonToVeLon(uint256 _lockDuration) public returns (uint256 tokenId) {
+        require(conversion, "conversion is not enabled");
+
+        uint256 userShare = balanceOf(msg.sender);
+        uint256 userLonBalance = userShare.mul(lonToken.balanceOf(address(this))).div(totalSupply());
```

Comment:
How about return this amount from `_redeemAndSendTo()` to save some gas (reading storage)?

Reply:
Good idea. I have added `returns (uint256 userTotalAmount)` for `_redeemAndSendTo()`.